frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -40,3 +43,9 @@ export const getPodLabels = (name: string) => {
     deploymentconfig: name,
   };
 };
+
+export const mergeData = (originalResource: K8sResourceKind, newResource: K8sResourceKind) => {
+  const mergedData = _.merge({}, originalResource || {}, newResource);
+  mergedData.metadata.labels = newResource.metadata.labels;
```

Comment:
You will have to do the same thing for template labels 
`mergedData.spec.template.metadata.labels = newResource.spec.template.metadata.labels`